src/parser/parse.rs

```diff
@@ -2120,83 +2121,85 @@ mod tests {
             }),
             (
-                r#"foo{label1="1" or label2="2" or label3="3" or label4="4"}"#,
+                r#"foo{label1="1" or label1="2" or label1="3" or label1="4"}"#,
+                {
+                    let matchers = Matchers::new(vec![]).with_or_matchers(vec![
+                        vec![Matcher::new(MatchOp::Equal, "label1", "1")],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "2")],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "3")],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "4")],
+                    ]);
+                    Expr::new_vector_selector(Some(String::from("foo")), matchers)
+                },
+            ),
+            (
+                r#"foo{label1="1" or label1="2" or label1="3", label2="4"}"#,
                 {
-                    let mut matchers = Matchers::new(vec![]).with_or_matchers(vec![vec![
-                        Matcher::new(MatchOp::Equal, "label1", "1"),
-                        Matcher::new(MatchOp::Equal, "label2", "2"),
-                        Matcher::new(MatchOp::Equal, "label3", "3"),
-                        Matcher::new(MatchOp::Equal, "label4", "4"),
-                    ]]);
-                    matchers.last_filter_action = FilterActionType::OR;
+                    let matchers = Matchers::new(vec![]).with_or_matchers(vec![
+                        vec![Matcher::new(MatchOp::Equal, "label1", "1")],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "2")],
+                        vec![
+                            Matcher::new(MatchOp::Equal, "label1", "3"),
+                            Matcher::new(MatchOp::Equal, "label2", "4"),
+                        ],
+                    ]);
+                    Expr::new_vector_selector(Some(String::from("foo")), matchers)
+                },
+            ),
+            (
+                r#"foo{label1="1", label2="2" or label1="3" or label1="4"}"#,
+                {
+                    let matchers = Matchers::new(vec![]).with_or_matchers(vec![
+                        vec![
+                            Matcher::new(MatchOp::Equal, "label1", "1"),
+                            Matcher::new(MatchOp::Equal, "label2", "2"),
+                        ],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "3")],
+                        vec![Matcher::new(MatchOp::Equal, "label1", "4")],
+                    ]);
                     Expr::new_vector_selector(Some(String::from("foo")), matchers)
                 },
             ),
         ];
         assert_cases(Case::new_result_cases(cases));
 
-        let promql = r#"a{label1="1" or label2="2"}"#;
-        let expected = r#"a{label1="1" or label2="2"}"#;
+        let promql = r#"a{label1="1"}"#;
+        let expected = r#"a{label1="1"}"#;
         let expr = parser::parse(promql).unwrap();
         assert_eq!(expr.to_string(), expected);
```

Comment:
We could use the `promql` as the expected string in the following tests.
```suggestion
        let promql = r#"a{label1="1"}"#;
        let expr = parser::parse(promql).unwrap();
        assert_eq!(expr.to_string(), promql);
```

Reply:
> We could use the `promql` as the expected string in the following tests.

It is indeed an unnecessary variable, I will fix it.